### PR TITLE
Predicting Early Hospital Readmission for Diabetic Patients.

### PR DESCRIPTION
**Pull request by TheUsefulNerd**
### Description
This PR introduces a complete Diabetes Readmission Prediction Model to the repository. The model is designed to predict the likelihood of diabetes patients being readmitted to the hospital based on various input features.

### Issue Resolved
This PR resolves issue #17 , implementing a comprehensive solution for diabetes readmission prediction.

### Checklist

- [x] My code follows the current [project structure](https://github.com/yashasvini121/predictive-calc/blob/master/docs/project-structure.md)
- [x] I have thoroughly reviewed and updated the `requirements.txt` file to include any new packages
- [x] The `predict.py` file includes a properly implemented `model_details()` function, or the notebook contains this function to print a detailed model report. **The model will not be accepted without this function**, as it is essential for generating the necessary model details.
- [ ] I have added relevant tests (if necessary).
- [x] I have added comments in the code where needed.
- [x] This PR is submitted under **Hacktoberfest**.
- [x] This PR is submitted under **GirlScript Summer of Code (GSSoC-Extd)**.

### CODE REVIEW

**Comment by TheUsefulNerd** [comment]
@yashasvini121, here it shows "this branch has conflicts that must be resolved." Is there something I can do?

**Comment by TheUsefulNerd** [comment]
The conflict was raise bcoz my code was extra in app.py. So, I just removed the markers so that my code add its urs. Is there anything else I can do?

**Comment by yashasvini121** [comment]
Your model is returning `0` for just every case. 

1. **Modify the return logic:**  
   In `model.py`, line 25, use an `if` statement to make the output more meaningful. For example:  
   ```python
   return "No" if prediction[0] == 0 else "Yes"
   ```

2. Improve the model

File: models/Diabetes_Readmission_Prediction/model.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        age_bins = {'10-20': 0, '20-30': 1, '30-40': 2, '40-50': 3, '50-60': 4, '60-70': 5, '70-80': 6, '80+': 7}
+        data['age'] = data['age'].map(age_bins)
+        return data
+
+    def predict(self, input_data: dict) -> int:
+        # Convert input_data to DataFrame
+        input_df = pd.DataFrame([input_data])
+        processed_data = self.preprocess(input_df)
+        prediction = self.model.predict(processed_data)
+        return prediction[0]
```
Comment (yashasvini121) [inline models/Diabetes_Readmission_Prediction/model.py:25]: **Modify the return logic:**  
   In `model.py`, line 25, use an `if` statement to make the output more meaningful. For example:  
   ```python
   return "No" if prediction[0] == 0 else "Yes"
   ```

**Review by yashasvini121 — CHANGES_REQUESTED** [review]
Your model is returning `0` for just every case.

File: models/Diabetes_Readmission_Prediction/model.py (hunk 1 above)
Comment (TheUsefulNerd) [reply models/Diabetes_Readmission_Prediction/model.py]: Modified it.

**Comment by TheUsefulNerd** [comment]
@yashasvini121 can you review the PR today?

**Review by yashasvini121 — CHANGES_REQUESTED** [review]
What I meant is that your model needs improvement because it's returning the same output (0) for every input combination.

**Comment by TheUsefulNerd** [comment]
Give me 2 days of time. I am learning undersampling and oversampling.  This time I will be focusing on getting a balanced recall. But, in the input there will be more that 25 to 30 features to be run by the model. It is because the category 2(minority class) has very less data compared to the majority class. But, I will try my best to enhance the performance.

**Pull request closed without merging**